rotkehlchen/api/v1/resources.py

```diff
@@ -2014,6 +2015,24 @@ def put(
         )
 
 
+class Eth2StakingEventsResource(BaseMethodView):
+    delete_schema = Eth2StakingEventsResetSchema()
+    put_schema = Eth2StakingEventsDecodingSchema()
+
+    @require_loggedin_user()
+    @use_kwargs(delete_schema, location='json')
+    def delete(self, entry_type: Literal[HistoryBaseEntryType.ETH_BLOCK_EVENT, HistoryBaseEntryType.ETH_WITHDRAWAL_EVENT]) -> Response:  # noqa: E501
+        return self.rest_api.reset_eth_staking_data(entry_type=entry_type)
+
+    @require_loggedin_user()
+    @use_kwargs(put_schema, location='json')
+    def put(self, async_query: bool, block_numbers: list[int]) -> Response:
```

Comment:
as I mentioned in the docs it probably makes sense to put this here as optional to redecode all?